lib/ymodel/base.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'ymodel/relatable'
+require 'ymodel/loadable'
+require 'ymodel/trigger'
+
+module YModel
+  # This is used in a similar manner to ActiveRecord
+  class Base
+    extend YModel::Relatable
+    extend YModel::Loadable
+    extend YModel::Trigger
+
+    def initialize(**record)
+      record.each do |k, v|
+        instance_variable_set '@' + k.to_s, v
```

Comment:
+1 on the string interpolation.  
I don't think verifying the attribute is that important. In fact I'd argue that leaving them around makes debugging a bit simples if someone would make a mistake. Its a bit besides the point because I don't think many YModels will be initialized.

As for the storage of data in Ivars. I think I started out with the idea of emulating active record much closer then I eventually did. I think ActiveRecord uses ivars, I'm pretty sure I've (ab)used them to in the past.